sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -357,3 +361,115 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+
+def test_missing_indicator():
+    X1_orig = np.array([
+        [-1,  -1,   1,   3],
+        [4,  -1,   0,  -1],
+        [8,  -1,   1,  0],
+        [0,  -1,   0,  15],
+        [16,  -1,   1,  19]
+    ])
+    X2_orig = np.array([
+        [5,  1,   1,   -1],
+        [-1,  -1,   2,  3],
+        [2,  3,   4,  0],
+        [0,  -1,   5,  -1],
+        [11,  -1,   1,  1]
+    ])
+
+    def assert_type(actual, is_sparse, sp, missing_values):
+        if sp is True:
+            assert_equal(actual, sparse.csc_matrix)
+        elif ((sp is "auto" and missing_values == 0) or sp is False):
```

Comment:
You should almost never use is for testing string equality. "Hello" is not "Hello world".split()[0]